modules/Walking_module/src/WalkingQPInverseKinematics.cpp

```diff
@@ -232,26 +455,255 @@ bool WalkingQPIK::setDesiredJointPosition(const iDynTree::VectorDynSize& regular
 }
 
 void WalkingQPIK::setDesiredFeetTransformation(const iDynTree::Transform& desiredLeftFootToWorldTransform,
-                                                    const iDynTree::Transform& desiredRightFootToWorldTransform)
+                                               const iDynTree::Transform& desiredRightFootToWorldTransform)
 {
     m_desiredLeftFootToWorldTransform = desiredLeftFootToWorldTransform;
     m_desiredRightFootToWorldTransform = desiredRightFootToWorldTransform;
 }
 
 void WalkingQPIK::setDesiredFeetTwist(const iDynTree::Twist& leftFootTwist,
-                                           const iDynTree::Twist& rightFootTwist)
+                                      const iDynTree::Twist& rightFootTwist)
 {
-    m_leftFootTwist = leftFootTwist;
-    m_rightFootTwist = rightFootTwist;
+    m_desiredLeftFootTwist = leftFootTwist;
+    m_desiredRightFootTwist = rightFootTwist;
+}
+
+void WalkingQPIK::setDesiredHandsTransformation(const iDynTree::Transform& desiredLeftHandToWorldTransform,
+                                                const iDynTree::Transform& desiredRightHandToWorldTransform)
+{
+    if(!m_enableHandRetargeting)
+        return;
+
+    m_desiredLeftHandToWorldTransform = desiredLeftHandToWorldTransform;
+    m_desiredRightHandToWorldTransform = desiredRightHandToWorldTransform;
+}
+
+void WalkingQPIK::setDesiredHandsTwist(const iDynTree::Twist& leftHandTwist,
+                                       const iDynTree::Twist& rightHandTwist)
+{
+    m_desiredLeftHandTwist = leftHandTwist;
+    m_desiredRightHandTwist = rightHandTwist;
 }
 
 void WalkingQPIK::setDesiredCoMVelocity(const iDynTree::Vector3& comVelocity)
 {
-    m_comVelocity = comVelocity;
+    m_desiredComVelocity = comVelocity;
 }
 
 void WalkingQPIK::setDesiredCoMPosition(const iDynTree::Position& desiredComPosition)
 {
     m_desiredComPosition = desiredComPosition;
 }
 
+void WalkingQPIK::evaluateHessianMatrix()
+{
+    // in that case the hessian matrix is related only to neck orientation and to
+    // the joint angle
+    auto hessianDense(iDynTree::toEigen(m_hessianDense));
```

Comment:
I am not sure if here ``auto`` is the safest choice. Maybe there could be ambiguity given that several Eigen could be initialized via map.

Reply:
Actually, here https://github.com/robotology/idyntree/issues/510#issuecomment-457219032, they suggested using `auto`

Reply:
True, I am not a super fan of the solution since also Eigen vectors could be initialized by copy from maps. At this point, I guess it would depend on the compiler to pick the most appropriate object, that probably will be the Eigen map since it is the best match. 
Yet I am not sure whether it is the safest option. Consider the case of ``const`` objects. Suppose ``m_hessianDense`` was ``const`` (it is not) and somewhere down in the code you modify ``hessianDense``. I am not sure whether in this case the compiler would turn ``hessianDense`` into a copy of ``m_hessianDense``, thus allocating memory to compile the code anyway. 
Anyway, I think it is fine in this case, but remember to use ``auto`` with caution, trying always to figure out what is the actual object instantiated.